packages/wallet/backend/src/socket/service.ts

```diff
@@ -0,0 +1,70 @@
+import { type Env } from '@/config/env'
+import { Logger } from 'winston'
+import socketIo from 'socket.io'
+import http from 'http'
+import { AccountService } from '@/account/service'
+import MessageType from './messageType'
+import { withSession } from '@/middleware/withSession'
+
+interface ISocketService {
+  init(args: http.Server): void
+  emitMoneyReceivedByUserId(id: string): void
+  emitMoneySentByUserId(id: string): void
+}
+
+interface SocketServiceDependencies {
+  logger: Logger
+  env: Env
+  accountService: AccountService
+}
+
+export class SocketService implements ISocketService {
+  private io: socketIo.Server | null = null
+
+  constructor(private deps: SocketServiceDependencies) {}
+
+  init(server: http.Server) {
+    this.io = new socketIo.Server(server, {
+      cors: {
+        origin: [
+          'http://localhost:4003',
+          `https://${this.deps.env.RAFIKI_MONEY_FRONTEND_HOST}`
+        ],
+        credentials: true
+      }
+    })
+    this.deps.logger.info(`Socket Server is started...`)
+
+    this.io.engine.use(withSession)
+    this.io.on('connection', async (socket) => {
+      const userId = socket.request.session.user.id
+
+      this.deps.logger.info(`A socket client ${userId} is connected...`)
+      socket.join(userId)
+
+      // Handle events
+      socket.on('disconnect', () => {
+        socket.leave(userId)
+        this.deps.logger.info(`A socket client ${userId} disconnected...`)
+      })
+
+      socket.on('error', (e) => {
+        this.deps.logger.error('Error in socket communication', e)
+      })
+    })
+  }
+
+  async emitMoneyReceivedByUserId(userId?: string) {
+    if (!userId) return
+
+    const accounts = await this.deps.accountService.getAccounts(userId, false)
+    this.io?.to(userId).emit(MessageType.MONEY_RECEIVED, accounts)
```

Comment:
I don't think the events should send all the accounts
The account id + updated balance should be enough
there are a few other options: send the account id (frontend will fetch it) or send the incoming payment id (frontend can fetch the incoming payment and can show with payment received money)

same applies to sent money event